openstack/clustering/v1/actions/results.go

```diff
@@ -0,0 +1,108 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// commonResult is the response of a base result.
+type commonResult struct {
+	gophercloud.Result
+}
+
+// GetResult is the response of a Get operations.
+type GetResult struct {
+	commonResult
+}
+
+// ActionPage contains a single page of all actions from a ListDetails call.
+type ActionPage struct {
+	pagination.LinkedPageBase
+}
+
+// Action represents a Detailed Action
+type Action struct {
+	Action       string                   `json:"action"`
+	Cause        string                   `json:"cause"`
+	CreatedAt    time.Time                `json:"-"`
+	Data         map[string]interface{}   `json:"data"`
+	DependedBy   []map[string]interface{} `json:"depended_by"`
+	DependsOn    []map[string]interface{} `json:"depends_on"`
+	StartTime    float32                  `json:"start_time"`
+	EndTime      float32                  `json:"end_time"`
+	ID           string                   `json:"id"`
+	Inputs       map[string]interface{}   `json:"inputs"`
+	Interval     int                      `json:"interval"`
+	Name         string                   `json:"name"`
+	Outputs      map[string]interface{}   `json:"outputs"`
+	Owner        string                   `json:"owner"`
+	ProjectUUID  string                   `json:"project"`
+	Status       string                   `json:"status"`
+	StatusReason string                   `json:"status_reason"`
+	Target       string                   `json:"target"`
+	Timeout      int                      `json:"timeout"`
+	UpdatedAt    time.Time                `json:"-"`
+	UserUUID     string                   `json:"user"`
+}
+
+// ExtractActions provides access to the list of actions in a page acquired from the List operation.
+func ExtractActions(r pagination.Page) ([]Action, error) {
+	var s struct {
+		Actions []Action `json:"actions"`
+	}
+	err := (r.(ActionPage)).ExtractInto(&s)
+	return s.Actions, err
+}
+
+// IsEmpty determines if a ClusterPage contains any results.
+func (page ActionPage) IsEmpty() (bool, error) {
+	actions, err := ExtractActions(page)
+	return len(actions) == 0, err
+}
+
+type JSONRFC3339Milli time.Time
+
+const RFC3339Milli = "2006-01-02T15:04:05.999999Z"
```

Comment:
Just remembered why have to use custom time parsing for JSONRFC3339Milli. Currently if using gophercloud.JSONRFC3339Milli and Senlin returns empty for the field, the parsing would fail. So I added this inside UnmarshalJSON for JSONRFC3339Milliif string(data) == "null" { *jt = JSONRFC3339Milli(time.Time{}) return nil } Should supporting empty for the time field be added into gophercloud.JSONRFC3339Milli or be in the results.go for clustering only?